esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2162,6 +2162,572 @@ bool Cws_accessEx::onResourcePermissions(IEspContext &context, IEspResourcePermi
 
     return true;
 }
+    
+bool Cws_accessEx::onQueryViews(IEspContext &context, IEspQueryViewsRequest &req, IEspQueryViewsResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+        
+        IArrayOf<IEspView> views;
+        StringArray names, descriptions;
+
+        try
+        {
+            secmgr->queryAllViews(names, descriptions);
+
+            ForEachItemIn(i, names)
+            {
+                Owned<IEspView> oneView = createView();
```

Comment:
createView() function is automatically generated by esp framework. It is generated based on the EspView structure made in ws_access.ecm file, and full definition is in ws_access.esp:
```
[build/generated/ws_access.esp]
extern "C"  IEspView *createView(const char *serv, const char *msgname){return ((IEspView *)new CView(serv /*, msgname*/));}
```